tests/unit/test_charm.py

```diff
@@ -181,6 +212,191 @@ def test_on_config_changed_iscsi(harness, mocker, iscsi_config):
+
+    The test is performed when the initiator name from initiatorname.iscsi contains
+    a different value from the one in the initiator-dictionary.
+    """
+    mocker.patch("charm.utils.is_container", return_value=False)
+    mock_getoutput = mocker.patch(
+        "charm.subprocess.getoutput", return_value="iqn.2020-07.canonical.com:lun1"
+    )
+    mocker.patch("charm.socket.getfqdn", return_value="testhost.testdomain")
+    mocker.patch("charm.subprocess.check_call")
+    mocker.patch("charm.subprocess.check_output")
+    mocker.patch("charm.StorageConnectorCharm._configure_deferred_restarts")
+    mocker.patch("charm.StorageConnectorCharm._defer_service_restart")
+    mocker.patch("charm.Path.chmod")
+
+    expected_initiator_content = dedent(
+        """\
+        ###############################################################################
+        # [ WARNING ]
+        # configuration file maintained by Juju
+        # local changes will be overwritten.
+        #
+        # DO NOT EDIT OR REMOVE THIS FILE!
+        # If you remove this file, the iSCSI daemon will not start.
+        # If you change the InitiatorName, existing access control lists
+        # may reject this initiator.  The InitiatorName must be unique
+        # for each iSCSI initiator.  Do NOT duplicate iSCSI InitiatorNames.
+        ###############################################################################
+        InitiatorName=iqn.2020-07.canonical.com:lun1"""
+    )
+
+    # configure the initiatorname.iscsi file with a different name from the one in
+    # the initiator-dictionary. The file should be rendered again from the name in
+    # the dictionary
+    initiator_content_with_different_name = dedent(
+        """\
+        ###############################################################################
+        # [ WARNING ]
+        # configuration file maintained by Juju
+        # local changes will be overwritten.
+        #
+        # DO NOT EDIT OR REMOVE THIS FILE!
+        # If you remove this file, the iSCSI daemon will not start.
+        # If you change the InitiatorName, existing access control lists
+        # may reject this initiator.  The InitiatorName must be unique
+        # for each iSCSI initiator.  Do NOT duplicate iSCSI InitiatorNames.
+        ###############################################################################
+        InitiatorName=new-iqn"""
+    )
+    harness.charm.ISCSI_INITIATOR_NAME.write_text(initiator_content_with_different_name)
+
+    mock_write_text = mocker.patch("charm.Path.write_text")
+
+    harness.update_config(iscsi_config)
+
+    assert harness.charm._get_initiator_name_from_file() == "new-iqn"
+
+    # assert that a random iqn isn't generated
+    assert call("/sbin/iscsi-iname") not in mock_getoutput.mock_calls
+    # check if file is being rendered with expected content
+    mock_write_text.assert_has_calls(
+        [
+            call(expected_initiator_content),
+        ]
+    )
```

Comment:
I think we don't need to validate the whole content. I only need to validate the configuration of InitiatorName. The content in the comment is not important. It will make the unit test more flexible when we do refactor or just fine tune the comment.

Reply:
I've managed to make changes which test only the required `InitiatorName` content instead of the whole file.